src/sage/rings/lazy_species.py

```diff
@@ -0,0 +1,1395 @@
+            [(E_3,), (X*E_2, 0), (X*E_2, 1), (X^3,)]
+
+            sage: from sage.rings.species import PolynomialSpecies
+            sage: L = LazySpecies(QQ, "X, Y")
+            sage: P = PolynomialSpecies(QQ, "X, Y")
+            sage: XY = L(P(PermutationGroup([], domain=[1, 2]), {0: [1], 1: [2]}))
+            sage: list((XY).isotypes(1, 1))
+            [(X*Y,)]
+
+            sage: list(E(XY).isotypes(2, 2))
+            [(E_2(X*Y),)]
+        """
+        multivariate = self.parent()._arity > 1
+        shape = tuple(shape)
+        for M, c in self[sum(shape)]:
+            if c not in ZZ or c < 0:
+                raise NotImplementedError("only implemented for proper non-virtual species")
+
+            if multivariate and tuple(M.grade()) != shape:
+                continue
+
+            if c == 1:
+                yield tuple([M])
+            else:
+                for e in range(c):
+                    yield (M, e)
+
+    def _test_isotypes(self, tester=None):
+        r"""
+        Check that isotypes and generating series are consistent.
+
+        TESTS::
+
+            sage: from sage.rings.species import PolynomialSpecies
+            sage: L = LazySpecies(QQ, "X, Y")
+            sage: P = PolynomialSpecies(QQ, "X, Y")
+            sage: XY = L(P(PermutationGroup([], domain=[1, 2]), {0: [1], 1: [2]}))
+            sage: XY._test_isotypes()
+        """
+        n = 3
+        P = self.parent()
+        if P._arity == 1:
+            s = list(self.isotypes(n))
+            assert len(s) == len(set(s)), f"isotypes for {n} are {s}, which is not a set"
+            coeff = self.isotype_generating_series()[n]
+            assert len(s) == coeff, f"the number of isotypes for {n} is {len(s)}, but the generating series gives {coeff}"
+        else:
+            shapes = IntegerVectors(n, length=P._arity)
+            for shape in shapes:
+                s = list(self.isotypes(*shape))
+                assert len(s) == len(set(s)), f"isotypes for {shape} are {s}, which is not a set"
+                coeff = self.isotype_generating_series()[n].coefficient(list(shape))
+                assert len(s) == coeff, f"the number of isotypes for {shape} is {len(s)}, but the generating series gives {coeff}"
+
+    def polynomial(self, degree=None, names=None):
+        r"""
+        Return ``self`` as a polynomial if ``self`` is actually so.
+
+        INPUT:
+
+        - ``degree`` -- ``None`` or an integer
+        - ``names`` -- names of the variables; if it is ``None``, the name of
+          the variables of the series is used
+
+        OUTPUT:
+
+        If ``degree`` is not ``None``, the terms of the series of
+        degree greater than ``degree`` are first truncated.  If
+        ``degree`` is ``None`` and the series is not a polynomial
+        polynomial, a ``ValueError`` is raised.
+
+        EXAMPLES::
+
+            sage: L = LazySpecies(ZZ, "X")
+            sage: E = L(lambda n: SymmetricGroup(n))
```

Comment:
```suggestion
            sage: E = L(SymmetricGroup)
```
A bit more concise way of constructing things. :)